test/ingest_search/ingest_search_test.go

```diff
@@ -0,0 +1,235 @@
+
+				var statusCmd strings.Builder
+				splunkBin := "/opt/splunk/bin/splunk"
+				username := "admin"
+				password := "$(cat /mnt/splunk-secrets/password)"
+				splunkCmd := "status"
+
+				fmt.Fprintf(&statusCmd, "%s %s -auth %s:%s", splunkBin, splunkCmd, username, password)
+				command := []string{"/bin/bash"}
+				stdin := statusCmd.String()
+				statusCmdResp, stderr, err := deployment.PodExecCommand(podName, command, stdin, false)
+				if err != nil {
+					testenvInstance.Log.Error(err, "Failed to execute command on pod", "pod", podName, "stdin", stdin, "statusCmdResp", statusCmdResp, "stderr", stderr)
+					return splcommon.PhaseError
+				}
+
+				if !strings.Contains(strings.ToLower(statusCmdResp), strings.ToLower("splunkd is running")) {
+					testenvInstance.Log.Error(err, "Failed to find splunkd running", "pod", podName, "statusCmdResp", statusCmdResp)
+					return splcommon.PhaseError
+				}
+
+				testenvInstance.Log.Info("Waiting for standalone splunkd status to be ready", "instance", standalone.ObjectMeta.Name, "Phase", standalone.Status.Phase)
+				return standalone.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Create an index
+			podName := fmt.Sprintf("splunk-%s-standalone-0", deployment.GetName())
+			indexName := "myTestIndex"
+
+			// Create an index on a standalone instance
+			err = testenv.CreateAnIndexStandalone(indexName, podName, deployment)
+			Expect(err).To(Succeed(), "Failed response to add index to splunk")
+
+			testenvInstance.Log.Info("Added index to Splunk", "instance", standalone.ObjectMeta.Name, "indexName", indexName, "Phase", standalone.Status.Phase)
+
+			logFile := "/tmp/test.log"
+
+			// Create a mock logfile to ingest
+			err = testenv.CreateMockLogfile(logFile, 1)
+			Expect(err).To(Succeed(), "Failed response to add index to splunk logfile %s", logFile)
+
+			// Copy log file and ingest it
+			err = testenv.IngestFileViaOneshot(logFile, indexName, podName, deployment)
+			Expect(err).To(Succeed(), "Failed to ingest logfile %s on pod %s", logFile, podName)
+
+			// Read first line to find a search token
+			var file, openErr = os.Open(logFile)
+			Expect(openErr).To(Succeed(), "Failed to open newly created logfile %s on pod %s", logFile, podName)
+
+			reader := bufio.NewReader(file)
+			var readErr error
+			firstLine, readErr = reader.ReadString('\n')
+			Expect(readErr).Should(Or(BeNil(), Equal(io.EOF)), "Failed to read first line of logfile %s on pod ", logFile, podName)
+
+			tokens := strings.Fields(firstLine)
+			Expect(len(tokens)).To(BeNumerically(">=", 2), "Incorrect tokens (%s) in first logline %s for logfile %s", tokens, firstLine, logFile)
+
+			searchToken := tokens[len(tokens)-1]
+			testenvInstance.Log.Info("Got search token successfully", "logFile", logFile, "searchToken", searchToken)
+
+			var searchString strings.Builder
+			fmt.Fprintf(&searchString, "index=%s | stats count by host", indexName)
```

Comment:
Nit: You could use fmt.sprintf for strings?

Reply:
Ya since were not appending/concatenating sprintf will work here and be just as efficient.  Switched.